src/Core/LaunchSites/LaunchSiteChecks.cs

```diff
@@ -348,23 +348,68 @@ public static void FindVesselsLandedAtKK(FlightState flightState, string landedA
+
+                                //Log.Normal("Vessel with distance: " + distance);
+                                if (distance == 0)
+                                {
+                                    Log.Normal("Vessel has same position as Launchsite, probably floating point error. Using gps checks instead.");
+
+                                    Vector2d tol = GetLatLonTolerance(launchSite.staticInstance.groupCenter.RefLatitude, launchSite.staticInstance.groupCenter.RefLongitude, maxDistance, body.Radius + launchSite.staticInstance.groupCenter.RadiusOffset);
+
+                                    if (Math.Abs(vessel.latitude - launchSite.staticInstance.groupCenter.RefLatitude) > tol.y || Math.Abs(vessel.longitude - launchSite.staticInstance.groupCenter.RefLongitude) > tol.x)
+                                    {
+                                        Log.Normal("Vessel is outside lat/lon tolerance");
+                                        Log.Debug($"Lat/Lon tolerance: {tol.y}, {tol.x}");
+                                        Log.Debug($"Vessel Lat/Lon: {vessel.latitude}, {vessel.longitude}");
+                                        Log.Debug($"Launchsite Lat/Lon: {launchSite.staticInstance.groupCenter.RefLatitude}, {launchSite.staticInstance.groupCenter.RefLongitude}");
+                                        vesselIndex++;
+                                        continue;
+                                    }
+                                    else
+                                    {
+                                        Log.Normal("Found Vessel at Launchsite with:");
+                                        Log.Debug($"Lat/Lon tolerance: {tol.y}, {tol.x}");
+                                        Log.Debug($"Vessel Lat/Lon: {vessel.latitude}, {vessel.longitude}");
+                                        Log.Debug($"Launchsite Lat/Lon: {launchSite.staticInstance.groupCenter.RefLatitude}, {launchSite.staticInstance.groupCenter.RefLongitude}");
+                                        count++;
+                                        name = vessel.vesselName;
+                                        idx = vesselIndex;
+                                        vType = vessel.vesselType;
+                                        break;
+                                    }
+                                }
+                                else if (distance < maxDistance)
+                                {
+                                    Log.Normal("Found Vessel at Launchsite with distance: " + distance);
+                                    count++;
+                                    name = vessel.vesselName;
+                                    idx = vesselIndex;
+                                    vType = vessel.vesselType;
+                                    break;
+                                }
+                            }
                         }
-                        //}
```

Comment:
This is becoming a horrible pile of 🍝. Refactor into a method. The other check below looks vaguely similar so perhaps try to unify with that, if possible.